File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    Enum.map(dna, fn neu -> transcript(neu) end)
  end

  defp transcript(dna) when dna == ?G, do: ?C
  defp transcript(dna) when dna == ?C, do: ?G
  defp transcript(dna) when dna == ?T, do: ?A
  defp transcript(dna) when dna == ?A, do: ?U
end

File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"

namespace rna_transcription
{
    char to_rna(char nucleotide)
    {
        if (nucleotide == 'C')
            return 'G';
        if (nucleotide == 'G')
            return 'C';
        if (nucleotide == 'A')
            return 'U';
        if (nucleotide == 'T')
            return 'A';

        return 0;
    }

    std::string to_rna(std::string nucleotides)
    {
        std::string complement = "";

        for (char n : nucleotides)
        {
            complement += to_rna(n);
        }

        return complement;
    }
} // namespace rna_transcription
